### ISSUE TITLE
Macros do not have their arguments set correctly

### ISSUE THREAD

**Issue by TDRRmk2** [zeta-group/ZDCode#1]
Marcos with parameters don't work at all in ZDCode 2.5.0. This is one example source ZDCode file:
```
class Testing
{
	macro ThisDoesntWork (i, d)
	{
		TNT1 A 0 A_SpawnItem(i, d);
	};

	label Spawn
	{
		inject ThisDoesntWork("Shotgun", 30);
	};
}
```

And the resulting DECORATE file is this:

```
Actor Testing
{

    States {
        Spawn:
            TNT1 A 0 A_SpawnItem(i, d)
    }
}
```

Notice how in the resulting DECORATE file the SpawnItem call uses the reference parameters, instead of replacing i with "Shotgun" and d with 30.

**Comment by wallabra** [comment]
An experimental fix is being pushed to the master branch and will be included in 2.5.1.

**Comment by wallabra** [comment]
The fix works as intended. Replacements now work.
